internal/uploader.go

```diff
@@ -62,6 +77,32 @@ func NewUploader(
 	return uploader
 }
 
+func NewSplitStreamUploader(
+	compressor compression.Compressor,
+	uploadingLocation storage.Folder,
+	partitions int,
+	blockSize int,
+) UploaderProvider {
+	if partitions <= 1 {
+		// Fallback to old implementation in order to skip unneeded steps:
+		return NewUploader(compressor, uploadingLocation)
+	}
+
+	uploader := &SplitStreamUploader{
+		Uploader: &Uploader{
```

Comment:
Just pass the `Uploader` to the `NewSplitStreamUploader` to avoid constructing the internal uploader.